x-pack/plugins/inference/server/tasks/nl_to_esql/actions/generate_esql.ts

```diff
@@ -5,77 +5,50 @@
  * 2.0.
  */
 
-import { Observable, map, merge, of, switchMap } from 'rxjs';
+import { map, tap, lastValueFrom } from 'rxjs';
 import type { Logger } from '@kbn/logging';
-import { ToolCall, ToolOptions } from '../../../../common/chat_complete/tools';
 import {
-  correctCommonEsqlMistakes,
-  generateFakeToolCallId,
+  ToolCall,
+  ToolOptions,
   isChatCompletionMessageEvent,
+  withoutTokenCountEvents,
+  withoutChunkEvents,
   Message,
   MessageRole,
-} from '../../../../common';
-import { InferenceClient, withoutTokenCountEvents } from '../../..';
-import { OutputCompleteEvent, OutputEventType } from '../../../../common/output';
+} from '@kbn/inference-common';
+import { correctCommonEsqlMistakes, generateFakeToolCallId } from '../../../../common';
 import { INLINE_ESQL_QUERY_REGEX } from '../../../../common/tasks/nl_to_esql/constants';
-import { EsqlDocumentBase } from '../doc_base';
-import { requestDocumentationSchema } from './shared';
-import type { NlToEsqlTaskEvent } from '../types';
-import type { FunctionCallingMode } from '../../../../common/chat_complete';
+import type { ActionsOptionsBase } from './types';
 
-export const generateEsqlTask = <TToolOptions extends ToolOptions>({
-  chatCompleteApi,
+export const generateEsqlTaskFn = <TToolOptions extends ToolOptions>({
```

Comment:
The changes in the `nl_to_esql` directory appear to be more substantial than (& unrelated to?) type/utility extraction. Can you help me understand the context for these changes?

Reply:
Oh right, I realized I did not mention anything on that part, sorry... 

There are two things:
1. I refactored the flow of the task to use a shared output observable instead of chaining observable calls (which was a nightmare to follow and to evolve, it's way easier to add steps and flow now)
2. I introduced a summarizing pre-step, to see the effect on generation efficiency, but the step is disabled by default and there's no way to enable it from production usages atm (it's only for testing via evaluation)
(3. I apparently also tweaked a few prompts, from evaluation tuning I did a while ago - the PR root is old)

That refactor is definitely something I want to merge, but tell me if we're not comfortable mixings things in that PR, I can extract those changes elsewhere.

Reply:
if it's not too much, maybe it's good to pull the changes to the ES|QL task out and into a separate PR, and then we can run the evaluation suites on both sides against it?

Reply:
Agreed. I actually forgot I used that other PR as base for those changes... I will revert that part

Reply:
Note for future me: `kbx-xxx-nl-to-esql-refactor-2`